packages/strategy-saml/src/server/graphql/mutation/assertSaml.ts

```diff
@@ -0,0 +1,398 @@
+const ATTRIBUTE_FIRST_NAME = "urn:oid:2.5.4.42";
+const ATTRIBUTE_LAST_NAME = "urn:oid:2.5.4.4";
+
+/**
+ * Finds an existing user by email address
+ * @param tx
+ * @param authority
+ * @param emailAddress
+ */
+async function findUserByEmail(
+  tx: DataLoaderExecutor<Pool | PoolClient>,
+  authority: SamlAuthority,
+  emailAddress: string
+): Promise<User | null> {
+  const emailAuthorityId = authority.details.emailAuthorityId;
+  if (!emailAuthorityId)
+    throw new Error("Expected emailAuthorityId to not be null");
+
+  const emailAuthority = await EmailAuthority.read(tx, emailAuthorityId);
+
+  const credential = await emailAuthority.credential(tx, emailAddress);
+
+  if (credential) {
+    return await User.read(tx, credential.userId);
+  } else {
+    return null;
+  }
+}
+
+async function createUserByEmail(
+  tx: PoolClient,
+  authority: SamlAuthority,
+  newAuthorizationId: string,
+  emailAddress: string,
+  userFullName: string
+): Promise<User> {
+  const emailAuthorityId = authority.details.emailAuthorityId;
+  if (!emailAuthorityId)
+    throw new Error("Expected emailAuthorityId to not be null");
+
+  const user = await User.write(
+    tx,
+    {
+      enabled: true,
+      id: v4(),
+      name: userFullName,
+      type: "human",
+    },
+    {
+      recordId: v4(),
+      createdAt: new Date(),
+      createdByAuthorizationId: newAuthorizationId,
+    }
+  );
+
+  const roles = await Promise.all(
+    authority.details.assignsCreatedUsersToRoleIds.map((id) =>
+      Role.read(tx, id)
+    )
+  );
+
+  const roleResults = await Promise.allSettled(
+    roles.map((role) =>
+      Role.write(
+        tx,
+        {
+          ...role,
+          userIds: [...role.userIds, user.id],
+        },
+        {
+          recordId: v4(),
+          createdByAuthorizationId: newAuthorizationId,
+          createdAt: new Date(),
+        }
+      )
+    )
+  );
+
+  for (const result of roleResults) {
+    if (result.status === "rejected") {
+      throw new Error(result.reason);
+    }
+  }
+
+  await EmailCredential.write(
+    tx,
+    {
+      authorityId: emailAuthorityId,
+      authorityUserId: emailAddress,
+      details: {},
+      enabled: true,
+      id: v4(),
+      userId: user.id,
+    },
+    {
+      recordId: v4(),
+      createdAt: new Date(),
+      createdByAuthorizationId: newAuthorizationId,
+    }
+  );
+
+  return user;
+}
+
+async function createSamlCredentialForUser(
+  tx: PoolClient,
+  authority: SamlAuthority,
+  user: User,
+  newAuthorizationId: string,
+  samlId: string
+): Promise<SamlCredential> {
+  return await SamlCredential.write(
+    tx,
+    {
+      userId: user.id,
+      id: v4(),
+      enabled: true,
+      details: {},
+      authorityUserId: samlId,
+      authorityId: authority.id,
+    },
+    {
+      recordId: v4(),
+      createdAt: new Date(),
+      createdByAuthorizationId: newAuthorizationId,
+    }
+  );
+}
+
+export const assertSaml: GraphQLFieldConfig<
```

Comment:
Is there any reason this is named `assertSaml` instead of `authenticateSaml` like the other strategies? It does generally seem to be equivalent to the others in terms of functionality.

Reply:
It's named that way because in SAML that type of endpoint is usually called "assertion" or something. However, it probably makes more sense to rename the GraphQL endpoint at least to be consistent with the others.